examples/arp_response.rs

```diff
@@ -126,39 +135,83 @@ fn main() {
         }
     }
 
-    pv::pv_close(nic);
+    pv_close(nic);
     println!("PV END");
 }
 
 fn process_packets(
-    nic: &mut PvNic,
-    packets: &mut Vec<PvPacket>,
+    nic: &mut Nic,
+    packets: &mut Vec<Packet>,
     batch_size: u32,
     src_mac_address: &MacAddr,
 ) -> u32 {
     let mut processed = 0;
     for i in (0..batch_size).rev() {
-        let payload_ptr = unsafe {
-            packets[i as usize]
-                .buffer
-                .add(packets[i as usize].start as usize)
-                .cast_const()
-        };
-
+        let packet_kind: PacketKind = get_packet_type(&packets[i as usize]);
         // analyze packet and create packet
-        unsafe {
-            if std::ptr::read(payload_ptr.offset(12)) == 0x08
-                && std::ptr::read(payload_ptr.offset(13)) == 0x06
-                && std::ptr::read(payload_ptr.offset(20)) == 0x00
-                && std::ptr::read(payload_ptr.offset(21)) == 0x01
-            // ARP request packet
-            {
-                gen_arp_response_packet(&mut packets[i as usize], src_mac_address);
-                processed += 1;
-            } else {
-                pv_free(nic, packets, i as usize);
+
+        let processing_result: Result<(), ()>;
```

Comment:
This variable is pointless. Remove it

Reply:
The variable moved into match branch instead of removing it in case that processing packet could fail